make-prs.py

```diff
@@ -0,0 +1,251 @@
+gh_client = github.Github(auth=auth)
+
+
+def get_diff_as_patch(repo: str, pull_request_number: int):
+    logger.info("Getting diff from PR ...")
+
+    pr_diff_url = gh_client.get_repo(repo).get_pull(pull_request_number).diff_url
+
+    res = requests.get(pr_diff_url)
+    if res.status_code != requests.codes.ok:
+        logger.error(f"Error getting diff from PR {pull_request_number}.")
+        exit(1)
+
+    with open(DIFF_FILE, "w") as f:
+        f.write(res.text)
+
+
+def clone(repo: str):
+    logger.info("Cloning repository ...")
+
+    shutil.rmtree(WORKING_DIR, ignore_errors=True)
+
+    clone_url = f"git@github.com:{repo}.git"
+    command = ["git", "clone", clone_url, WORKING_DIR]
+
+    result = subprocess.run(command, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error cloning reop {repo}: {result.stderr}")
+        exit(1)
+
+
+def create_new_branch_based_on(base: str) -> str:
+    logger.info("Creating branch ...")
+
+    head = f"{base}-update"
+
+    command = [
+        "git",
+        "checkout",
+        "-b",
+        head,
+        f"origin/{base}",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error creating new branch {head}: {result.stderr}")
+        exit(1)
+
+    return head
+
+
+def apply_patch() -> bool:
+    logger.info("Applying patch ...")
+
+    command = [
+        "git",
+        "apply",
+        "--3way",
+        "../diff.patch",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        if "conflicts" in result.stderr:
+            return True
+        else:
+            logger.error(f"Error applying diff: {result.stderr}")
+            exit(1)
+
+
+def commit_diff(head: str):
+    logger.info("Pushing changes ...")
+
+    add = ["git", "add", "."]
+    subprocess.run(add, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    commit = ["git", "commit", "-m", "chore: merging diff"]
+    subprocess.run(commit, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    push = ["git", "push", "-f", "origin", head]
+    result = subprocess.run(push, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        logger.error(f"Error pushing changes to branch {head}: {result.stderr}")
+        exit(1)
+
+
+def create_pr(repo: str, base: str, head: str, conflict: bool):
+    logger.info("Creating PR ...")
+
+    title = f"chore: merging diff into branch {base}"
+    body = f"Automated change: merging diff into branch {base}"
+
+    if conflict:
+        title = f"{title} CONFLICTS!"
+        body = f"**Conflicts! Need human intervention!**\n\n{body}"
+
+    repo = gh_client.get_repo(repo)
+    pr = repo.create_pull(base=base, head=head, title=title, body=body)
+    if not pr:
+        logger.error("Error creating PR.")
```

Comment:
Is there any error message here? Wouldn't a failure be an exception? (In which case we can probably just let it bubble up for a script like this)

Reply:
I had a look at the code here, I think the `if not pr` part is not necessary, because `repo.create_pull` would raise an exception if `status >= 400`. I will remove this check and this line of log here.